third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -543,6 +544,55 @@ impl ValueImpl {
 
         Ok(res)
     }
+
+    fn compare(&self, other: &Self) -> PartialVMResult<Ordering> {
+        use ValueImpl::*;
+
+        let res = match (self, other) {
+            (U8(l), U8(r)) => l.cmp(r),
+            (U16(l), U16(r)) => l.cmp(r),
+            (U32(l), U32(r)) => l.cmp(r),
+            (U64(l), U64(r)) => l.cmp(r),
+            (U128(l), U128(r)) => l.cmp(r),
+            (U256(l), U256(r)) => l.cmp(r),
+            (Bool(l), Bool(r)) => l.cmp(r),
+            (Address(l), Address(r)) => l.cmp(r),
+
+            (Container(l), Container(r)) => l.compare(r)?,
+
+            (ContainerRef(l), ContainerRef(r)) => l.compare(r)?,
+            (IndexedRef(l), IndexedRef(r)) => l.compare(r)?,
+
+            // Disallow equality for delayed values. The rationale behind this
+            // semantics is that identifiers might not be deterministic, and
+            // therefore equality can have different outcomes on different nodes
+            // of the network. Note that the error returned here is not an
+            // invariant violation but a runtime error.
+            (DelayedFieldID { .. }, DelayedFieldID { .. }) => {
+                return Err(PartialVMError::new(StatusCode::VM_EXTENSION_ERROR)
+                    .with_message("cannot compare delayed values".to_string()))
+            },
+
+            (Invalid, _)
+            | (U8(_), _)
+            | (U16(_), _)
+            | (U32(_), _)
+            | (U64(_), _)
+            | (U128(_), _)
+            | (U256(_), _)
+            | (Bool(_), _)
+            | (Address(_), _)
+            | (Container(_), _)
+            | (ContainerRef(_), _)
+            | (IndexedRef(_), _)
+            | (DelayedFieldID { .. }, _) => {
+                return Err(PartialVMError::new(StatusCode::INTERNAL_TYPE_ERROR)
+                    .with_message(format!("cannot compare values: {:?}, {:?}", self, other)))
```

Comment:
Shall we use `Display` instead of `Debug` in the message (hopefully that means more stable error message)?

Reply:
this is copied from equals. I am not sure what is more stable from MoveVM prespective, but seems like this should be the same as in equals. 

and changing equals would require feature flag? if we wanted to change eeverything to {} , we should do that separately, I think

Reply:
okay.. putting Debug text to messages we dare not to change generally make me very sad but you don't make it worse I guess

Reply:
@vgao1996 for thoughts

Reply:
The error message of VMError is (more or less unfortunately, making it hard to show to users) not stored on chain, so I do not think it is very important to be stable.